compute/volumes.go

```diff
@@ -0,0 +1,216 @@
+}
+
+type Volume struct {
+	ID             string   `json:"id"`
+	Name           string   `json:"name"`
+	Owner          string   `json:"owner_uuid"`
+	Type           string   `json:"type"`
+	FileSystemPath string   `json:"filesystem_path"`
+	State          string   `json:"state"`
+	Networks       []string `json:"networks"`
+	Refs           []string `json:"refs"`
+}
+
+type ListVolumesInput struct {
+	Name  string
+	Size  string
+	State string
+	Type  string
+}
+
+func (c *VolumesClient) List(ctx context.Context, input *ListVolumesInput) ([]*Volume, error) {
+	fullPath := path.Join("/", c.client.AccountName, "volumes")
+
+	query := &url.Values{}
+	if input.Name != "" {
+		query.Set("name", input.Name)
+	}
+	if input.Size != "" {
+		query.Set("size", input.Size)
+	}
+	if input.State != "" {
+		query.Set("state", input.State)
+	}
+	if input.Type != "" {
+		query.Set("type", input.Type)
+	}
+
+	reqInputs := client.RequestInput{
+		Method: http.MethodGet,
+		Path:   fullPath,
+		Query:  query,
+	}
+	respReader, err := c.client.ExecuteRequest(ctx, reqInputs)
+	if respReader != nil {
+		defer respReader.Close()
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list volumes")
+	}
+
+	var result []*Volume
+	decoder := json.NewDecoder(respReader)
+	if err = decoder.Decode(&result); err != nil {
+		return nil, errors.Wrap(err, "unable to decode list volumes response")
+	}
+
+	return result, nil
+}
+
+type CreateVolumeInput struct {
+	Name     string
+	Size     int64
+	Networks []string
+	Type     string
+}
+
+func (input *CreateVolumeInput) toAPI() map[string]interface{} {
+	result := make(map[string]interface{}, 0)
+
+	if input.Name != "" {
+		result["name"] = input.Name
+	}
+
+	if input.Size != 0 {
+		result["size"] = input.Size
+	}
+
+	if input.Type != "" {
+		result["type"] = input.Type
+	}
+
+	if len(input.Networks) > 0 {
+		result["networks"] = input.Networks
+	}
+
+	return result
+}
+
+func (c *VolumesClient) Create(ctx context.Context, input *CreateVolumeInput) (*Volume, error) {
+	fullPath := path.Join("/", c.client.AccountName, "volumes")
+
+	reqInputs := client.RequestInput{
+		Method: http.MethodPost,
+		Path:   fullPath,
+		Body:   input.toAPI(),
+	}
+	respReader, err := c.client.ExecuteRequest(ctx, reqInputs)
+	if respReader != nil {
+		defer respReader.Close()
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create volume")
+	}
+
+	var result *Volume
+	decoder := json.NewDecoder(respReader)
+	if err = decoder.Decode(&result); err != nil {
+		return nil, errors.Wrap(err, "unable to decode create volume response")
+	}
+
+	return result, nil
+}
+
+type DeleteVolumeInput struct {
+	ID string
+}
+
+func (c *VolumesClient) Delete(ctx context.Context, input *DeleteVolumeInput) error {
+	fullPath := path.Join("/", c.client.AccountName, "volumes", input.ID)
+	reqInputs := client.RequestInput{
+		Method: http.MethodDelete,
+		Path:   fullPath,
+	}
+	response, err := c.client.ExecuteRequestRaw(ctx, reqInputs)
```

Comment:
`s/response/resp/g` to be consistent with nearly all other users of `req`/`resp`. ?